service/web/stream/manager.go

```diff
@@ -107,23 +109,69 @@ func (s *Manager) CreateNewStream(server metrov1.Subscriber_StreamingPullServer,
 }
 
 // Acknowledge ...
-func (s *Manager) Acknowledge(ctx context.Context, req *ParsedStreamingPullRequest) error {
-	for _, ackMsg := range req.AckMessages {
+func (s *Manager) Acknowledge(ctx context.Context, parsedReq *ParsedStreamingPullRequest) error {
+
+	// holds a map of ackMsgs to their corresponding originating server addresses
+	msgsToBeProxied := make(map[string][]*subscriber.AckMessage, 0)
+
+	for _, ackMsg := range parsedReq.AckMessages {
 		if ackMsg.MatchesOriginatingMessageServer() {
 			// find active stream
 			if pullStream, ok := s.pullStreams[ackMsg.SubscriberID]; ok {
 				pullStream.acknowledge(ctx, ackMsg)
 			}
 		} else {
-			// proxy request to the correct server
+			proxyAddr := ackMsg.ServerAddress
+			if _, ok := msgsToBeProxied[proxyAddr]; !ok {
+				// init empty slice
+				msgsToBeProxied[proxyAddr] = make([]*subscriber.AckMessage, 0)
+			}
+			msgsToBeProxied[proxyAddr] = append(msgsToBeProxied[proxyAddr], ackMsg)
 		}
 	}
 
+	if len(msgsToBeProxied) > 0 {
```

Comment:
can we refactor this part into a common function, seems duplicated below

Reply:
done.